repo_guide.md

```diff
@@ -0,0 +1,46 @@
+
+## The DAG-Poseidon Webserver
+
+We have a webserver running, which currently has two possible APIs implemented:
+
+1. Under [https://c107-224.cloud.gwdg.de/packages](https://c107-224.cloud.gwdg.de/packages), one receives a JSON-list of packages available at the server. Each JSON entry has a title, a description, a last-modified date and a package version.
+2. Under [https://c107-224.cloud.gwdg.de/package_table](https://c107-224.cloud.gwdg.de/package_table), one receives a HTML table of packages with download links, available at the server.
+3. Under [https://c107-224.cloud.gwdg.de/zip_file/\<package_name\>](https://c107-224.cloud.gwdg.de/zip_file/<package_name>), a zip-file for the given package can be downloaded, which contains the package content.
+
+The webserver is powered by an HTTP server program called `poseidon-http-server`, whose source-code can be reviewed at the [same repository](https://github.com/poseidon-framework/poseidon-hs) as `trident` is sourced. For those interested to run their own instance of this webserver, for example under `localhost`, the server comes with a short command line help. It gets installed via `stack install`, similarly to `trident`. The program first scans the given directories for Poseidon packages, then creates zip-files for them, and then starts a HTTP server listening - by default - to port 3000, and providing the two APIs listed above.
+
+## Contributing to our central repositories
+
+The Poseidon framework has a strongly decentralized philosophy and relies very much on a community of users willing to prepare and improve the data in the repositories. To ensure a professional and welcoming atmosphere we would first of all like to ask you to respect our [Contributor Code of Conduct](conduct.md) in all interactions with the Poseidon team and other users on Github and beyond.
+
+There are two basic ways you can contribute to Poseidon:
+
+1) If you identify a mistake in any package metadata, be it context data (`.janno`-files), package-meta-data (`POSEIDON.yml`) or bibliographic information (`.bib` files), we welcome contributions to correct or extend that data. This goes most easily through our github-package, which you can fork, commit changes and ask for Pull requests. You can also use the issue tracker on github (for which you can find some help [here](https://lab.github.com/githubtraining/introduction-to-github)). See below for more details on this.
+
+2) If you have new genetic data to share with us, we would like to process that data ourselves, to ensure maximum compatibility with the existing database and to minimise batch effects. So if you want us to add an entirely new package with genetic data, please share with us the raw data, such as BAM or fastq-files, so that we can generate the data and include it into a new Poseidon package.
```

Comment:
We should add that we would like to reprocess the genotype data, but some of the context data in the janno files can very well be prepared right away by the user who submits the request in this scenario. It's not trivial to prepare non-minimal .janno files, and can take a good amount of time.

Reply:
Ah OK. I thought the first point covers that, but indeed that covers only existing packages. OK, will improve that.